full-service/src/service/ledger.rs

```diff
@@ -176,25 +181,28 @@ where
         Ok(self.ledger_db.contains_key_image(key_image)?)
     }
 
-    fn get_network_fees(&self) -> BTreeMap<TokenId, u64> {
-        let mut fees = self
+    fn get_network_fees(&self) -> Result<BTreeMap<TokenId, u64>, LedgerServiceError> {
+        // Get the last block information from all nodes we are aware of, in parallel.
+        let last_block_infos = self
             .peer_manager
             .conns()
             .par_iter()
-            .filter_map(|conn| conn.fetch_block_info(empty()).ok())
-            .map(|block_info| block_info.minimum_fees)
-            .reduce(BTreeMap::new, |mut acc, fees| {
-                for (token_id, fee) in fees {
-                    acc.entry(token_id)
-                        .and_modify(|e| *e = cmp::max(*e, fee))
-                        .or_insert(fee);
-                }
-                acc
-            });
-        fees.entry(Mob::ID)
-            .and_modify(|e| *e = cmp::max(*e, Mob::MINIMUM_FEE))
-            .or_insert(Mob::MINIMUM_FEE);
-        fees
+            .filter_map(|conn| conn.fetch_block_info(std::iter::empty()).ok())
+            .collect::<Vec<_>>();
+
+        // Ensure that all nodes agree on the minimum fee map.
+        if last_block_infos
+            .windows(2)
+            .any(|window| window[0].minimum_fees != window[1].minimum_fees)
+        {
+            return Err(LedgerServiceError::InconsistentLastBlockInfo);
+        }
+
+        // Return the fee map.
+        last_block_infos
+            .get(0)
+            .map(|info| info.minimum_fees.clone())
+            .ok_or(LedgerServiceError::NoLastBlockInfo)
     }
 
     fn get_network_block_version(&self) -> BlockVersion {
```

Comment:
Should this also do the same as the above?

Reply:
That's a tricky one. There's an edgecase where a node could be behind for whatever reason, just at the time where a new block version was introduced. So expecting all nodes to agree on the block version could mean returning an intermittent error. I am not sure there's a right answer here, but I think the options are:
1) Erroring when all the available nodes do not agree on the block version, with the assumption that this should be a temporary condition
2) Choosing the most common block version at the time the query is made. I am leaning towards this option.
Thoughts?

Reply:
Maybe the highest, since the rest _should_ be catching up to the one that has the highest?

Reply:
Unless there is a malicious node that's setting the BlockVersion higher than the rest, causing it to always submit to that one

Reply:
I think the intermittent error is acceptable, honestly, since how often are we assuming this is going to happen?

Reply:
Pushed a change to treat it similarly to the minimum_fee map